packages/Php70/src/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector.php

```diff
@@ -0,0 +1,259 @@
+            return $this->getParametersForExpr($node->name, $scope);
+        }
+
+        return $this->getParametersForName($node->name, $node->args, $scope);
+    }
+
+    /**
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForExpr(Expr $expr, Scope $scope): array
+    {
+        $type = $scope->getType($expr);
+
+        if (! $type instanceof ParametersAcceptor) {
+            return [];
+        }
+
+        return $type->getParameters();
+    }
+
+    /**
+     * @param Node\Arg[] $args
+     *
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForName(Name $name, array $args, Scope $scope): array
+    {
+        try {
+            return ParametersAcceptorSelector::selectFromArgs(
+                $scope,
+                $args,
+                $this->broker->getFunction($name, $scope)->getVariants()
+            )->getParameters();
+        } catch (FunctionNotFoundException $functionNotFoundException) {
+            return [];
+        }
+    }
+
+    private function processArgument(Expr $expr, Scope $scope): VariableAssignPair
+    {
+        if (
+            (
+                $expr instanceof Assign
+                || $expr instanceof AssignRef
+                || $expr instanceof AssignOp
+            )
+            && $this->isVariableLikeNode($expr->var)
+        ) {
+            return new VariableAssignPair($expr->var, $expr);
+        }
+
+        $variable = new Variable($this->getVariableNameFor($expr, $scope));
+
+        return new VariableAssignPair($variable, new Assign($variable, $expr));
+    }
+
+    private function isVariableLikeNode(Node $node): bool
+    {
+        return $node instanceof Variable
+            || $node instanceof ArrayDimFetch
+            || $node instanceof PropertyFetch
+            || $node instanceof StaticPropertyFetch;
+    }
+
+    private function getVariableNameFor(Expr $expr, Scope $scope): string
+    {
+        if ($expr instanceof New_ && $expr->class instanceof Name) {
+            $name = $this->classNaming->getShortName($expr->class);
+        } else {
+            $name = $this->getName($expr);
+        }
+
+        return lcfirst($this->createCountedValueName($name, $scope));
+    }
+
+    private function createCountedValueName(?string $valueName, Scope $scope): string
+    {
+        $countedValueName = $valueName ?? self::DEFAULT_VARIABLE_NAME;
+
+        // make sure variable name is unique
+        if (! $scope->hasVariableType($countedValueName)->yes()) {
+            return $countedValueName;
+        }
+
+        // we need to add number suffix until the variable is unique
+        $i = 2;
+        $countedValueNamePart = $countedValueName;
+        while ($scope->hasVariableType($countedValueName)->yes()) {
+            $countedValueName = $countedValueNamePart . $i;
+            ++$i;
+        }
+
+        return $countedValueName;
+    }
+}
```

Comment:
This would be great if abstracted in `AbstractRector`, as this logic is used in multiple places now

Reply:
@TomasVotruba How would you handle things like suffix ? (`ForRepeatedCountToOwnVariableRector` uses a suffix)

Reply:
What do you mean? Suffix of what?

I'd just move this to protected method to abstract class, so it can be re-used in multiple Rector rules.